### PR TITLE
Display ads: Use `promote.keywords_from_context` to generate keywords

### PR DESCRIPTION
**Pull request by dwick**
:eyeglasses: @bsimpson63 

This was the change that caused memoize to fail, already been reviewed for the code itself, just wanted to make sure nothing else unforeseen happens.

related: https://github.com/reddit/reddit-public/pull/4478

### CODE REVIEW

File: reddit_adzerk/adzerkads.py [hunk 1]
```diff
@@ -22,18 +24,17 @@ class Ads(BaseAds):
...
+            include_subscriptions=False,
+        )
 
         # adzerk reporting is easier when not using a space in the tag
-        if isinstance(c.site, DefaultSR):
-            site_name = FRONTPAGE_NAME
+        if Frontpage.name in keywords:
+            keywords = keywords - {Frontpage.name} | {FRONTPAGE_NAME}
 
-        sr = quote(site_name.lower())
```
Comment (bsimpson63) [inline reddit_adzerk/adzerkads.py]: looks like you're no longer lowercasing all the subreddit names

**Comment by bsimpson63** [comment]
:nail_care: 

Seems ok otherwise.

Now that the lookup is stalecached we should almost never be hitting the "real" cache. It might be nice to have a mechanism for local-only memoization/caching for cases like this.

Did you consider adding an attribute to the `Subreddit` object indicating whether it was nsfw for advertising purposes? That seems like a more direct way to do what you want and wouldn't be too big of a pain because collections don't change often.

File: reddit_adzerk/adzerkads.py (hunk 1 above)
Comment (dwick) [inline reddit_adzerk/adzerkads.py]: this wasn't needed: http://help.adzerk.com/hc/en-us/articles/202624319-Keyword-Targeting

**Comment by dwick** [comment]
> Did you consider adding an attribute to the Subreddit object indicating whether it was nsfw for advertising purposes

This is something we want to do on a larger scale eventually. "ads categories".. I'd rather not have to do it right now though

:haircut:

**Comment by bsimpson63** [comment]
:cry: ok

:fish:

**Pull request merged**